### DISCUSSION TITLE
Controlling the index.html file generation for a React application in Nx Workspace

### DISCUSSION THREAD

**Discussion by rpspall** [nrwl/nx#14795] [General]
Migrating an application from create-react-application (CRA) to an Nx workspace...

There are significant differences in the way that CRA generates its index.html and Nx workspace generates it's index.html.

Obviously, both take your application index.html as the base and then generate the production index.html using an algorithm of sorts...

CRA replaces "%PUBLIC_URL% with your homepage variable value (package.json) in your original index.html. This is very helpful for locating various resources, and you can do something similar with Nx workspaces using custom Nx environment variables (must begin with NX_ prefix). CRA also loads its built modules using the same homepage path.

NX doesn't have the concept of a homepage in the project.json and instead uses a baseHref in the project.json. NX then uses the baseHref value to add a html <base href="????"> element to the generated index.html.

Unfortunately (and I could go into the long version with details if necessary) <base href="????"> has an impact on relative pathing (well fortunately if it is what you want!) for any resources loaded downstream by your application!

Bottom line... Is it possible to customize the way that NX generates its index.html, so it uses a similar model to CRA, namely no html <base href="????"> element and instead use the baseHref (or may renamed to homepage) variable to append in front of paths in the generated index.html file?
